security/security.c

```diff
@@ -28,12 +28,26 @@
 #include <linux/string.h>
 #include <linux/msg.h>
 #include <net/flow.h>
+#include <linux/static_call.h>
+#include <linux/loop_unrolling.h>
+#include <linux/build_bug.h>
 
 #define MAX_LSM_EVM_XATTR	2
 
 /* How many LSMs were built into the kernel? */
 #define LSM_COUNT (__end_lsm_info - __start_lsm_info)
 
+#define STATIC_SLOT(HOOK, NUM)	security_static_slot_##HOOK##_##NUM
+
+/*
+ * Static slots are placeholders for potential LSM hooks.
+ * Instead of a costly indirect call, they use static calls.
+ */
+#define SECURITY_STATIC_SLOT_COUNT 11
+static_assert(SECURITY_STATIC_SLOT_COUNT <= MAX_UNROLL_MACRO_LOOP);
+#define SECURITY_FOREACH_STATIC_SLOT(M, ...)				\
+	UNROLL_MACRO_LOOP(SECURITY_STATIC_SLOT_COUNT, M, __VA_ARGS__)
```

Comment:
yeah this is where I feel we should not bother, atleast initially about creating a general API out of this macro, let's keep it limited to security.

Keeping this in mind, you can probably name the header file lsm_static_calls.h we can revert to a generic api if people ask for it.